pkg/actor/actor.go

```diff
@@ -79,95 +74,19 @@ type Actor interface {
-	conditionVersionCheckedTrue := condition.True(api.CrdbVersionChecked, conditions)
-	conditionVersionCheckedFalse := condition.False(api.CrdbVersionChecked, conditions)
-	conditionCertificateGeneratedTrue := condition.True(api.CertificateGenerated, conditions)
-
-	var actorsToExecute []Actor
-
-	if featureDecommissionEnabled && conditionInitializedTrue {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.DecommissionAction])
-	}
-
-	if featureVersionValidatorEnabled && conditionVersionCheckedFalse && (conditionInitializedTrue || conditionInitializedFalse) {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.VersionCheckerAction])
-	}
-
-	if !conditionCertificateGeneratedTrue {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.GenerateCertAction])
-	}
-
-	if featureVersionValidatorEnabled && conditionVersionCheckedTrue && conditionInitializedTrue {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.PartitionedUpdateAction])
-	} else if !featureVersionValidatorEnabled && conditionInitializedTrue {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.PartitionedUpdateAction])
-	}
-
-	if featureResizePVCEnabled && conditionInitializedTrue {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.ResizePVCAction])
-	}
-
-	if featureVersionValidatorEnabled && conditionVersionCheckedTrue && (conditionInitializedTrue || conditionInitializedFalse) {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.DeployAction])
-	} else if !featureVersionValidatorEnabled && (conditionInitializedTrue || conditionInitializedFalse) {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.DeployAction])
-	}
-
-	if featureVersionValidatorEnabled && conditionVersionCheckedTrue && conditionInitializedFalse {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.InitializeAction])
-	} else if !featureVersionValidatorEnabled && conditionInitializedFalse {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.InitializeAction])
-	}
-
-	// TODO: conditionVersionCheckedTrue should probably be contingent on featureVersionValidatorEnabled, like with other actions
-	if featureClusterRestartEnabled && conditionVersionCheckedTrue && (conditionInitializedTrue || conditionInitializedFalse) {
-		actorsToExecute = append(actorsToExecute, cd.actors[api.ClusterRestartAction])
-	}
-
-	return actorsToExecute
-}
-
-func newAction(atype string, scheme *runtime.Scheme, cl client.Client) action {
+func newAction(scheme *runtime.Scheme, cl client.Client, config *rest.Config, clientset kubernetes.Interface) action {
 	return action{
-		client: cl,
-		scheme: scheme,
+		client:    cl,
+		clientset: clientset,
+		scheme:    scheme,
+		config:    config,
 	}
 }
 
 // action is the base set of common parameters required by other actions
 type action struct {
-	client client.Client
-	scheme *runtime.Scheme
+	client    client.Client
+	clientset kubernetes.Interface
+	scheme    *runtime.Scheme
+	config    *rest.Config
```

Comment:
If we've got a clientset, do we need scheme and/or config?

Reply:
Unfortunately they do seem to be used sporadically throughout the Actors...maybe it's possible to refactor them away but I think that should be done in a later PR (if at all)?